packages/@react-aria/grid/src/useGridCell.ts

```diff
@@ -53,14 +69,20 @@ export interface GridCellAria {
 export function useGridCell<T, C extends GridCollection<T>>(props: GridCellProps, state: GridState<T, C>, ref: RefObject<FocusableElement | null>): GridCellAria {
   let {
     node,
+    focusMode,
     isVirtualized,
-    focusMode = 'child',
+    isEditable = false,
     shouldSelectOnPressUp,
     onAction
   } = props;
 
   let {direction} = useLocale();
-  let {keyboardDelegate, actions: {onCellAction}} = gridMap.get(state);
+  let {id, keyboardDelegate, keyboardNavigationBehavior, actions: {onCellAction}} = gridMap.get(state);
+
+  let isEditing = state.editingKey === node.key;
+  let isReadOnly = state.isReadOnly || props.isReadOnly;
+
+  focusMode ??= isEditable || keyboardNavigationBehavior === 'tab' ? 'cell' : 'child';
```

Comment:
Do we want to move onto the first child automatically when in tab navigation mode?

Reply:
thanks for the clarification! The "table edit mode with tab navigation example" story demonstrates the scenario you are describing right? If so, I think what you have now is fine good it always takes an explicit key press other than the arrow keys to enter a cell regardless of its edibility.